mexico-city/src/managers/chihuahua_manager.rs

```diff
@@ -201,6 +202,9 @@ fn dispatch_on_program(
     protocol_state: &ProtocolState,
     colima::Program { code, .. }: colima::Program,
 ) -> ProvisioningResult {
+    // Buffer the program, it will be used in batch process
+    PROG_AND_DATA_BUFFER.lock()?.buffer_program(code.as_slice());
```

Comment:
We should probably use `expect()` with an explicit error message if the process crashes due to not being able to obtain the lock, here.

Reply:
The protocol_state also returns an error instead of raising panic when falling to obtain the lock. Do you propose to use panic/expect for unrecoverable error? My thought here is to use less panic as possible. Lock error is propagated via automatically wrapping the error message, `format!({:?},err)` into our custom error type.

Reply:
I think you are right: the runtime should never panic, instead an error should be produced and propagated upwards.